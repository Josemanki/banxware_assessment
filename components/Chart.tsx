import React from 'react';

import { THistoricBalance } from '../types';

import { Line } from 'react-chartjs-2';
import {
  Chart as ChartJS,
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend,
} from 'chart.js';

// Registers necessary components for the chart.
ChartJS.register(CategoryScale, LinearScale, PointElement, LineElement, Title, Tooltip, Legend);

type Props = {
  chartData: THistoricBalance[];
};

const Chart = ({ chartData }: Props) => {
  const labels = chartData.map((item) => item.timestamp);
  const historicBalance = chartData.map((item) => item.historicBalance);

  const data = {
    labels: labels,
    datasets: [
      {
        label: 'Balance',
        backgroundColor: '#42f5a1',
        borderColor: '#42f5a1',
        data: historicBalance,
      },
    ],
  };

  return (
    <div>
      <Line data={data} />
    </div>
  );
};

export default Chart;
