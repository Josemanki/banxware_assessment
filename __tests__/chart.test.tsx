import { render } from '@testing-library/react';
import Chart from '../components/Chart';
import '@testing-library/jest-dom';

const chartData = [
  {
    timestamp: '30-5-2022',
    historicBalance: 9109,
  },
  {
    timestamp: '29-5-2022',
    historicBalance: 9543,
  },
  {
    timestamp: '28-5-2022',
    historicBalance: 9991,
  },
  {
    timestamp: '27-5-2022',
    historicBalance: 8322,
  },
];

test('Should render the Chart component', () => {
  const { asFragment } = render(<Chart chartData={chartData} />);
  expect(asFragment()).toMatchSnapshot();
});
