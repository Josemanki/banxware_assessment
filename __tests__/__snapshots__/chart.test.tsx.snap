// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the Chart component 1`] = `
<DocumentFragment>
  <div>
    <canvas
      height="150"
      role="img"
      width="300"
    />
  </div>
</DocumentFragment>
`;
